x/leveragelp/keeper/msg_server_close_positions.go

```diff
@@ -35,8 +50,25 @@ func (k msgServer) ClosePositions(goCtx context.Context, msg *types.MsgClosePosi
 		if !found || err != nil {
 			continue
 		}
-		_, _ = k.ClosePositionIfUnderStopLossPrice(ctx, &position, pool, ammPool)
+		under, early, err := k.ClosePositionIfUnderStopLossPrice(ctx, &position, pool, ammPool)
+		if under && !early {
+			leftToClose--
+		} else if err != nil {
+			// Add log about error or not closed
+			closeLog = append(closeLog, "Position: Address:%s Id:%s cannot be liquidated due to err: %s", position.Address, strconv.FormatUint(position.Id, 10), err.Error())
+		} else {
+			closeLog = append(closeLog, "Position: Address:%s Id:%s is not under stop loss", position.Address, strconv.FormatUint(position.Id, 10))
+		}
 	}
 
+	ctx.EventManager().EmitEvent(sdk.NewEvent(types.EventClosePositions,
+		sdk.NewAttribute("liquidations_total", strconv.Itoa(len(msg.Liquidate))),
+		sdk.NewAttribute("liquidations_not_liquidated", strconv.Itoa(leftToLiquidate)),
+		sdk.NewAttribute("liquidations", strings.Join(liqLog, "\n")),
+		sdk.NewAttribute("stop_loss", strings.Join(closeLog, "\n")),
+		sdk.NewAttribute("stop_loss_total", strconv.Itoa(len(msg.Liquidate))),
+		sdk.NewAttribute("stop_loss_not_closed", strconv.Itoa(leftToClose)),
```

Comment:
isn't the data stop_loss_not_closed and liquidations_not_liquidated redundant as both can be counted using liquidations and stop_loss attributes

Reply:
what do you mean ? these values indicates positions that were not closed/liquidated due to any reason.

Let's say 10 positions were passed and 2 were not liquidated, so we are adding it to log

Reply:
am I missing something ?

Reply:
number of closeLog == stop_loss_not_closed
and
number of liqLog == liquidations_not_liquidated

Also, liquidations_total and stop_loss_total are also redundant as these are tx events and this data is already available in tx.

It doesn't harm to keep it unless this too many of this tx is done

Reply:
Then it would be taking storage in block events, and if there is no use, it is better not to keep it.

Reply:
ah, I'm fine with this too, added this for simplicity on logs

Reply:
removed